app/models/valid_fulfilled_at_date.rb

```diff
@@ -0,0 +1,56 @@
+class ValidFulfilledAtDate
+
+  include ActiveModel::Validations
+  include DateHelper
+
+  validate :valid_format
+  validate :not_in_future, if: :to_time
+  validate :recent, if: :to_time
+
+  def self.min
+    # Beginning of previous fiscal year
```

Comment:
Okay, maybe I'm missing something, but why the previous fiscal year and not the current fiscal year? Wouldn't the previous fiscal year already be closed?